static/js/services/lineage-model-generator.js

```diff
@@ -63,13 +63,37 @@ angular.module('inboxServices').factory('LineageModelGenerator',
         });
     };
 
+    var mergeParents = function(doc, lineage) {
+      var current = doc;
+      lineage.forEach(function(hydrated) {
+        if (!current) {
+          return;
+        }
+        if (hydrated) {
+          current.parent = hydrated;
+        }
+        current = current.parent;
+      });
+      return doc;
+    };
+
     return {
-      contact: function(id) {
+
+      /**
+       * Fetch a contact and its lineage by the given uuid. Returns a
+       * contact model, or if options.merge is true the doc with the
+       * lineage inline.
+       */
+      contact: function(id, options) {
```

Comment:
I'm personally at war with the `options` pattern, because I think it devolves into giant opaque spaghetti options. So I'd expose a separate `mergedContact` (or something) function. 

But if that's just my recent trauma, then go ahead with `options`. :) Your choice.

Reply:
Yeah I understand your concern. From the caller side I find options much easier to understand, eg:

`contact(id, { merge: true })` is somewhat self documenting, whereas:
`contact(id, true)` is useless.

As long as it's documented on the function what the options are then I prefer it overall. Might be worth raising this for discussion at the next iteration meeting?